File: checkout-core/src/main/java/com/adyen/checkout/core/handler/RedirectHandler.java
/*
 * Copyright (c) 2018 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by timon on 13/07/2018.
 */

package com.adyen.checkout.core.handler;

import android.support.annotation.NonNull;

import com.adyen.checkout.core.RedirectDetails;

public interface RedirectHandler {
    /**
     * Called when a redirect is required to continue with the payment.
     *
     * @param redirectDetails The details of the redirect.
     */
    void onRedirectRequired(@NonNull RedirectDetails redirectDetails);
}

File: redirect/src/main/java/com/adyen/checkout/redirect/handler/RedirectHandler.kt
/*
 * Copyright (c) 2022 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by josephj on 18/8/2022.
 */
package com.adyen.checkout.redirect.handler

import android.content.Context
import android.net.Uri
import com.adyen.checkout.core.exception.CheckoutException
import org.json.JSONObject

interface RedirectHandler {
    /**
     * A redirect may return to the application using the ReturnUrl when properly setup in an Intent Filter. Is usually contains result information
     * as parameters on that returnUrl. This method parses those results and returns a [JSONObject] to be used in the details call.
     *
     * @param data The returned Uri
     * @return The parsed value to be passed on the payments/details call, on the details parameter.
     */
    @Throws(CheckoutException::class)
    fun parseRedirectResult(data: Uri?): JSONObject

    fun launchUriRedirect(context: Context, url: String?)
}
